src/libpriv/rpmostree-kernel.cxx

```diff
@@ -105,13 +105,13 @@ find_kernel_and_initramfs_in_bootdir (int rootfs_dfd, const char *bootdir, char
   return TRUE;
 }
 
-/* Given a directory @subpath, find the first child that is a directory,
- * returning it in @out_subdir.  If there are multiple directories,
+/* Given a directory @subpath, find the first child that is a directory
+ * and contains a `vmlinuz` file, returning it in @out_subdir.  If there are multiple directories,
  * return an error.
  */
 static gboolean
-find_ensure_one_subdirectory (int rootfs_dfd, const char *subpath, char **out_subdir,
-                              GCancellable *cancellable, GError **error)
+find_dir_with_vmlinuz (int rootfs_dfd, const char *subpath, char **out_subdir,
```

Comment:
We actually need to make this an official ostree API - also this code desperately wants to be in Rust.

Reply:
https://github.com/ostreedev/ostree-rs-ext/pull/361